### PR TITLE
#2378 show RA their event's status or pending depending on date

### PR DESCRIPTION
**Pull request by stenington**
This should handle #2378, showing RAs their event status as "pending" prior to a finalization date, and showing "official/unofficial" afterwards. The finalization date is set by `DATES_RPE_OFFICIALITY_FINALIZED_*` variables, which will be set to April 22, 2020 in production.

The way I have it written, I believe RPE statuses will be considered finalized on April 23, starting at midnight, Pacific time.

### CODE REVIEW

File: app/javascript/ra/events/Event.js [hunk 1]
```diff
@@ -23,6 +23,12 @@ export default function (event) {
 
   this.capacity = event.capacity || null;
 
+  if (event.officiality_finalized) {
+    this.officiality = event.official ? "official" : "unofficial";
+  } else {
+    this.officiality = "pending";
+  }
+
```
Comment (shaunxp20) [inline app/javascript/ra/events/Event.js]: Not a blocker, just wondering if it makes sense to move this logic to `regional_pitch_event.rb`, and just expose one method/JSON item for `event.status`? 🤔 😅 😬

Comment (stenington) [reply app/javascript/ra/events/Event.js]: Yeah, I could be talked into that.

**Comment by stenington** [comment]
@shaunxp20 maybe this looks a little better?

**Comment by shaunxp20** [comment]
> @shaunxp20 maybe this looks a little better?

Oh nice! I like how it makes it easier to reuse this functionality now too, it's not tied to the Vue.js layer. Looks good to me! 👍 🙌

**Comment by shaunxp20** [comment]
> This one's looking good to me! 🌞 Thanks for digging into our time zones! 🥇 Btw, I only ran the RPE specs (all passing for me), I didn't manually test it, but the changes look good. 👍

Gah, comment/approval for the wrong PR! 🤦‍♂ 😄 😂

**Pull request merged**